### PR TITLE
Fix @unix_only and @windows_only depreciation

### PR DESCRIPTION
**Pull request by laurentheirendt**

### CODE REVIEW

File: src/cpx_common.jl [hunk 1]
```diff
@@ -1,11 +1,15 @@
 # makes calling C functions a bit easier
 macro cpx_ccall(func, args...)
     f = "CPX$(func)"
-    @unix_only return quote
-        ccall(($f,libcplex), $(args...))
+    @static if is_unix()
```
Comment (mlubin) [inline src/cpx_common.jl]: This case doesn't need ``@static``

File: src/cpx_common.jl [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+        return quote
+            ccall(($f,libcplex), $(args...))
+        end
     end
-    @windows_only return quote
-        ccall(($f,libcplex), stdcall, $(args...))
+    @static if is_windows()
```
Comment (mlubin) [inline src/cpx_common.jl]: Same here, no ``@static``

**Comment by laurentheirendt** [comment]
This was the only thing that muted the following warnings in Julia 0.5.0-rc3+0.

julia> using CPLEX
INFO: Precompiling module CPLEX...
WARNING: `@unix_only` is deprecated, use `@static if is_unix()` instead
 in depwarn(::String, ::Symbol) at ./deprecated.jl:64
 in depwarn(::String, ::Symbol) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in @unix_only(::Any) at ./deprecated.jl:493
 in include_from_node1(::String) at ./loading.jl:426
 in include_from_node1(::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in include_from_node1(::String) at ./loading.jl:426
 in include_from_node1(::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in macro expansion; at ./none:2 [inlined]
 in anonymous at ./<missing>:?
 in eval(::Module, ::Any) at ./boot.jl:234
 in eval(::Module, ::Any) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in process_options(::Base.JLOptions) at ./client.jl:239
 in _start() at ./client.jl:318
 in _start() at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
while loading ~/.julia/v0.5/CPLEX/src/cpx_common.jl, in expression starting on line 2

How do you suggest of fixing this?

**Comment by mlubin** [comment]
Just use ``is_unix()`` instead of ``@static is_unix()``. The ``@static`` does nothing useful in this particular case because the code runs at compilation time anyway.

File: src/cpx_common.jl [hunk 3]
```diff
@@ -14,7 +18,7 @@ macro cpx_ccall_intercept(model, func, args...)
     quote
         ccall(:jl_exit_on_sigint, Void, (Cint,), convert(Cint,0))
         ret = try
-            $(@windows? :(ccall(($f,libcplex), stdcall, $(args...))) : :(ccall(($f,libcplex), $(args...))) )
+            $(is_windows() ? :(ccall(($f,libcplex), stdcall, $(args...))) : :(ccall(($f,libcplex), $(args...))) )
```
Comment (mlubin) [inline src/cpx_common.jl]: ``@static`` should stay here because it's runtime code

**Comment by mlubin** [comment]
Thanks for this. I'll merge if you confirm that you've tested the code locally on both 0.4 and 0.5.

**Comment by laurentheirendt** [comment]
You are welcome. Yes, I tested it on v0.4.5 and v0.5.0-rc3+0. Thanks.

**Pull request merged**